src/web/mod.rs

```diff
@@ -188,7 +208,80 @@ impl Handler for CratesfyiHandler {
                 }
 
                 Self::chain(&self.pool_factory, err).handle(req)
-            })
+            });
+
+        let path: &[&str] = &req.url.path();
+        match path {
```

Comment:
Why did you choose this approach?

I'm not convinced this is the way to go, as it requires maintenance every time we add a route if we want it to be tracked. I'm thinking the recorder struct could be changed into an iron handler:

```rust
struct RequestRecorder {
    handler: Box<dyn Handler>,
    route_name: String,
}

impl Handler for RequestRecorder {
    fn handle(&self, request: &Request) -> IronResult<Response> {
        let start = Instant::now();
        let result = self.handler.handle(request);
        let elapsed = start.elapsed();

        // Record the request and the elapsed time

        result
    }
}
```

Then, in `src/web/routes.rs`, the methods to add static, internal and rustdoc routes are changed to wrap the handlers in `RequestRecorder`. We'd also get the route name for free if we assume "static" and "rustdoc" for static and rustdoc routes, while using the `pattern` as the route name for internal routes.